text/0000-named-blocks.md

```diff
@@ -0,0 +1,730 @@
+that intends to address similar use cases, so there is some risk of
+introducing an API that doesn't fit in with what the rest of the world
+is doing.
+
+### Syntax highlighting changes
+
+Some syntax highlighters might have trouble with this syntax; all
+the editors I've tried it on look reasonable, but GitHub's Handlebars
+parser isn't too kind (hence I've been using `html` snippets instead
+of `handlebars` snippets):
+
+```hbs
+<x-modal>
+  <@header as |c|>
+    ...
+  </@header>
+
+  <@main as |c|>
+    ...
+  </@main>
+</x-modal>
+```
+
+### Conditionally passing blocks?
+
+This RFC does NOT introduce any kind of facility for conditionally passing
+blocks, e.g.:
+
+```html
+{{! this syntax is INVALID! }}
+<x-layout>
+  <@header>...</@header>
+  <@main>...</@main>
+
+  {{#if userCanProceed}}
+    <@footer>
+      {{submit-button}}
+    </@footer>
+  {{/if}}
+</x-layout>
+```
+
+This might be desirable in the future, particularly for use cases
+involving flex-ish layouts where the component changes behavior /
+appearance based on whether blocks on passed in.
+
+# Alternatives
+
+I'd proposed a JSX-y [attr/component-centric](https://github.com/emberjs/rfcs/pull/203) syntax for passing what are essentially DOM lambdas, rendered with `{{component}}`. Perhaps we'll add something like that feature in the future, but it's a much less natural enhancement to Ember than named blocks.
+
+# Considerations for Future RFCs
+
+## Defining Default Blocks
+
+There's not really a nice way defining default blocks inside your
+component layout (i.e. the block you render when known is provided at
+invocation time), but then again I believe the following would be
+a workable approach that is probably support by the features proposed in
+this RFC?
+
+```
+{{#with-blocks}}
+  <@mainOrDefault>
+    {{#if main}}
+      {{main}}
+    {{else}}
+      I am the default main block when none is passed in.
+    {{/if}}
+  </@mainOrDefault>
+
+  <@footerOrDefault>
+    {{#if footer}}
+      {{footer}}
+    {{else}}
+      I am the default footer block when none is passed in.
+    {{/if}}
+  </@footerOrDefault>
+
+  <@render as |@mainOrDefault @footerOrDefault|>
+    {{! this specially-named block gets passed all the other blocks above}}
+
+    {{mainOrDefault}}
+    {{footerOrDefault}}
+  </@render>
+{{/with-blocks}}
+```
+
+Either way, it feels hacky and weird and I would be surprised if we'd
+want/need a future RFC to define a nicer way to support default blocks.
+
+## Allow passing multiple blocks with the same name
+
+e.g.
+
+```html
+{{#power-select}}
+  <@option value="foo">
+    <em>Foo</em>
+  </@option>
+  <@option value="bar">
+    <blink>Bar</blink>
+  </option>
+{{/power-select}}
+```
+
+This RFC defines that passing multiple blocks with the same name is a
+syntax error, but it's something we might want to relax in the future
+for certain cases where you want to pass arrays of blocks, such as
+`power-select` or use cases involving tables.
```

Comment:
Does this hold true for multiple invocation at runtime? `{{#each items as |item|}}  <@option> </@option> {{/each}}`